### PR TITLE
Add WebGL extension constants

### PR DESCRIPTION
**Pull request by Diggsey**

### CODE REVIEW

File: khronos_api/build.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        let path = entry.path();
+        let ext_name = path.file_name().unwrap().to_str().unwrap();
+        if path.is_dir() && ext_name != "template" {
+            let ext_path = path.join("extension.xml");
+            if ext_path.is_file() {
+                writeln!(file, "&*include_bytes!({:?}),", ext_path.to_str().unwrap()).unwrap();
+            }
+        }
+    }
+    writeln!(file, "]").unwrap();
```
Comment (brendanzab) [inline khronos_api/build.rs]: Why is this needed? Perhaps a small comment could help make it clearer, along with pointing to the `WEBGL_EXT_XML` constant, where this will be used?

Comment (brendanzab) [reply khronos_api/build.rs]: Oh, saw the comment in https://github.com/brendanzab/gl-rs/pull/447#issuecomment-363089825 now - I think it still could use some added info though.

File: khronos_api/src/lib.rs [hunk 2]
```diff
@@ -34,3 +34,6 @@ pub const WEBGL_IDL: &'static [u8] = include_bytes!("../api_webgl/specs/latest/1
 
 /// The contents of [`webgl2.idl`](https://github.com/KhronosGroup/WebGL/blob/master/specs/latest/2.0/webgl2.idl)
 pub const WEBGL2_IDL: &'static [u8] = include_bytes!("../api_webgl/specs/latest/2.0/webgl2.idl");
+
+/// The contents of the WebGL extension XML files
+pub const WEBGL_EXT_XML: &'static [&'static [u8]] = include!(concat!(env!("OUT_DIR"), "/webgl_exts.rs"));
```
Comment (brendanzab) [inline khronos_api/src/lib.rs]: Any reason why you decided to go with this, instead of `&[(&str, &[u8])]`?

Comment (Diggsey) [reply khronos_api/src/lib.rs]: I was originally going to extract the IDL in the `khronos_api` crate, but I decided that it would be better to keep the `khronos_api` crate as simple as possible, and just expose the raw XML. That way if I (or someone else) wants something else from the XML, then it's available. Given that I'm storing the whole XML, there's no need to store the name separately.

File: khronos_api/build.rs (hunk 1 above)
Comment (Diggsey) [reply khronos_api/build.rs]: Yeah, I'll add some comments explaining what's going on here - is there any other information you think would be worth adding?

Comment (brendanzab) [reply khronos_api/build.rs]: Nope, that should be fine! I think folks should get the gist of it from the code once they are in the right frame of mind.

**Comment by Diggsey** [comment]
I've updated with comments and some stuff I missed.

File: khronos_api/src/lib.rs [hunk 3]
```diff
@@ -34,3 +34,7 @@ pub const WEBGL_IDL: &'static [u8] = include_bytes!("../api_webgl/specs/latest/1
 
 /// The contents of [`webgl2.idl`](https://github.com/KhronosGroup/WebGL/blob/master/specs/latest/2.0/webgl2.idl)
 pub const WEBGL2_IDL: &'static [u8] = include_bytes!("../api_webgl/specs/latest/2.0/webgl2.idl");
+
+/// The contents of the WebGL extension XML files
+/// These are discovered via a build script to avoid having to list each extension by name.
```
Comment (brendanzab) [inline khronos_api/src/lib.rs]: Minor nit: Probably don't want the second line showing up in the user facing documentation, so we can probably skip this.

**Comment by Diggsey** [comment]
Fixed.

**Comment by brendanzab** [comment]
Awesme!

**Comment by brendanzab** [comment]
I think I'll probably wait to deploy this once you give me the go-ahead with the webgl stuff!
